carma-messenger-core/cpp_message/src/MobilityOperation_Message.cpp

```diff
@@ -73,10 +73,14 @@ namespace cpp_message
             header.recipient_id=recipient_id;
             
             //get bsm id
+            //sender_bsm_id is meant to represent the vehicle BSM id in hex string (Ex: FFFFFFFF)
+            //Since we're decoding this from a binary value to a MobilityHeader value, the transition would be from Binary to Hex
             str_len=message->value.choice.TestMessage03.header.hostBSMId.size;
+            std::string temp;
             for(size_t i=0;i<str_len;i++){
-                sender_bsm_id +=message->value.choice.TestMessage03.header.hostBSMId.buf[i];
+                temp +=message->value.choice.TestMessage02.header.hostBSMId.buf[i];
```

Comment:
This seems like an incorrect change to me. According to the ASN.1 spec for the mobility messages the MobilityOperation message is TestMessage3. Since this is the MobilityOperation_Message.cpp file I would expect only that message to be referenced.
Sample of the spec where this is defined. 
```
TestMessage00 ::= SEQUENCE {
   -- TestMessage00 is only for MobilityRequest message
   header          MobilityHeader,
   body            MobilityRequest,
   ...
   }
TestMessage01 ::= SEQUENCE {
   -- TestMessage01 is only for MobilityResponse message
   header    MobilityHeader,
   body      MobilityResponse,
   ...
   }
TestMessage02 ::= SEQUENCE {
   -- TestMessage02 is only for MobilityPath message
   header    MobilityHeader,
   body      MobilityPath,
   ...  
   }
TestMessage03 ::= SEQUENCE {
   -- TestMessage03 is only for MobilityOperation message
   header    MobilityHeader,
   body      MobilityOperation,
   ...  
   }
```